docs/nima/builder.adoc

```diff
@@ -0,0 +1,210 @@
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Helidon Builder
+:description: about Helidon Builder
+:keywords: helidon, java, microservices, reactive, virtual threads, config, builder-config
+
+
+
+
+== Contents
+
+- <<Overview, Overview>>
+- <<Maven Coordinates, Maven Coordinates>>
+- <<Usage, Usage>>
+- <<Configuration, Configuration>>
+- <<Examples, Examples>>
+
+//the purpose of this article is to introduce Helidon Builder and provide some high-level information about how and why its used. The examples and eventual Builder guide will provide the detailed steps. 
+
+
+== Overview
+
+Helidon Builder is an annotation processor that code-generates fluent builder support just by adding `@Builder` to your declared Java types. This declarative style of using annotations helps you by reducing the amount of boilerplate [redundant?] code you need to write. 
+
+Specifically, Helidon Builder offers a rich set of capabilities, including:
+
+* Extensibility:  Helidon Builder is designed to be customized and extended in order to integrate directly with the Helidon Configuration system.  
+
+* Visitor Pattern: Helidon Builder allows every attribute in the @Builder-annotated target type to be visited for your general use.
+
+* Validation: Having the visitor pattern available allows some convenient mix-in behaviors, one of which is checking for required attributes that must be set as part of the build() method of your target. By default, all non-Optional attributes are considered to be required attributes. Helidon Builder provides validation out of the box in the generated implementation classes.
+
+* Interception:  You can provide other custom semantics during the build() method (similar to validation) for other types of interception or decoration on your target builders.
+
+* Packaging and usage choice: Helidon Builder can be used as a standalone module in your build as long as you are using Java 11 or later. When used in a standalone manner in your build, you can use the `requireLibraryDependencies()` attribute to toggle between supporting libraries (like validation). This allows you to either generate code directly into your application, or have it supplied by Helidon libraries. This gives you an opportunity to use Helidon's tooling in isolation in your project build, even if you don't plan to use Helidon in any other part of your application runtime. 
+
+//Do we want to say you don't plan to use Helidon??? 
+
+== Maven Coordinates
+
+To enable Builder, add the following dependencies and annotation processor path to your project's `pom.xml`:
+
+.Builder dependencies
+[source,java]
+
+----
+<dependency>
+   <groupId>io.helidon.builder</groupId>
+   <artifactId>helidon-builder</artifactId>
+   <version>4.0.0-ALPHA4</version>
```

Comment:
ALPHA6 (or just prepare for the M1 version)

Reply:
This should be defined as a variable in the shared dependencies file. I'll make sure we have the correct version attributes set for go-live.